Blog/app/controllers/application_controller.rb

```diff
@@ -2,4 +2,17 @@
 
 class ApplicationController < ActionController::Base
   before_action :authenticate_user!, unless: :devise_controller?
+  def owner?(user_id)
+    return if user_id == current_user.id
+
+    redirect_to articles_path, notice: "Action not allowed!!"
+  end
+
+  def owner_object?(data)
+    if data.nil? || data.user_id.nil?
```

Comment:
Should works also

```suggestion
    unless data&.user_id
```

Reply:
I changed it to 'unless data&.user_id' and, while saving the file, my rubocop-code-monitoring automatically changed it to 'if data&.user_id', which works of course.